src/eigen_value.c

```diff
@@ -0,0 +1,44 @@
+/******************************************************
+* Author : Amuleen Gulati
+* Function : eigen value
+* Application : This function computes the eigen values vector having degree n for 
+* an input matrix[n][n] (the SDM matrix)
+* Input : number of sensors n as integer and sd_matrix[n][n] of doubles 
+* Output : vector of size n of eigen vectors
+*/
+
+#include "..\include\eigen.h"
+
+double* eigen_value(int n, double** D){
+/*Initialize an array to store eigen values */
+double* evalue;
+evalue = (double*) malloc(n*sizeof(double));
+
+/* Create a GSL matrix M */
+gsl_matrix* M = gsl_matrix_alloc(n, n);
+
+/* Assign the elements of Matrix */
+for (int i = 0; i < n; i++){
+    for (int j = 0; j < n; j++){
+        gsl_matrix_set(M, i, j, D[i][j]);
+    }
+}
+
+/* Create a GSL matrix to store eigenvectors */
+gsl_matrix* evecs = gsl_matrix_alloc(n,n);
+
+/* Create a GSL vector to store eigenvalues */
+gsl_vector* evals = gsl_vector_alloc(n);
```

Comment:
Need to free.

Reply:
I think you might not have noticed, but I have deallocated memory at the bottom of the function. I just put it in comments because I was not sure about the exact location where the memory should be deallocated. I have fixed this now.